library/Mockery/Expectation.php

```diff
@@ -558,6 +558,28 @@ public function andReturnUsing(...$args)
         return $this;
     }
 
+    /**
+     * Sets up a closure to return the nth argument from the expected method call
+     *
+     * @param int $index
+     * @return self
+     */
+    public function andReturnArgument($index)
+    {
+        if (!is_int($index) || $index < 0) {
+            throw new \InvalidArgumentException("Invalid argument index supplied. Index must be a positive integer.");
+        }
+        $closure = function (...$args) use ($index) {
+            if (isset($args[$index])) {
+                return $args[$index];
+            }
+            throw new Exception("Cannot return an argument value. No argument exists for the index $index");
```

Comment:
I'm not 100% sure if throwing an exception here is correct. 

What do people think is the right behaviour when the argument index is out-of-bounds for a particular method call?

Reply:
I think it's a reasonable starting point. Could throw a `OutOfBoundsException` to be more explicit.

Reply:
makes sense